gradle/code-coverage.gradle

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+apply plugin: 'jacoco'
+
+repositories {
+  mavenCentral()
```

Comment:
Do we need mavenCentral? 
What are we really pulling from Maven?

Reply:
If `jacoco` is declared in `build.gradle` of root project, the repository declaration is required, because there is no other repositories mentioned. However I can try to find a way to declare `jacoco` in `buildSrc`, and apply it to the root project.

Reply:
As my comment on above, seems the repository declaration is required in root `build.gradle` if applying `jacoco` in the root project, but I didn't find any info to prove that.
Although the configuration and "task" registrations in this `gradle/code-coverage.gradle` file can move into `buildSrc` directory, I think `jacoco` plugin should be applied to the root project, because it comply with our instinct that these tasks generate the overall code coverage report.
Actually I'm confused what code of "build" configuration should be put into `buildSrc` directory, or the more put there the better, but I could refactor them to a better place in the future.